### ISSUE TITLE
logcli emits "proto: duplicate proto type registered: ingester.Series"

### ISSUE THREAD

**Issue by daurnimator** [grafana/loki#3104]
**Describe the bug**

Every `logcli` invocation spits out the warning/message "proto: duplicate proto type registered: ingester.Series"

e.g.

```
$ logcli query '{job="ircd/irc"}' --limit 1
2020-12-28 15:54:11.369278 I | proto: duplicate proto type registered: ingester.Series
https://grafana.hashbang.sh/api/datasources/proxy/1/loki/api/v1/query_range?direction=BACKWARD&end=1609131251369783025&limit=1&query=%7Bjob%3D%22ircd%2Firc%22%7D&start=1609127651369783025
Common labels: {app="irc", container="irc", filename="/var/log/pods/ircd_irc-deployment-58f6bbd5cf-4rlc5_7c9735be-8f94-4640-8b5f-fd00963316aa/irc/0.log", job="ircd/irc", namespace="ircd", node="hashbang-3hxb9", pod="irc-deployment-58f6bbd5cf-4rlc5", pod_template_hash="58f6bbd5cf", stream="stderr"}
2020-12-28T15:54:08+11:00 {} 2020-12-28T04:54:08.915Z : debug : quit       : * is no longer on the server
```

```
$ logcli --version
2020-12-28 15:53:28.632516 I | proto: duplicate proto type registered: ingester.Series
logcli, version 2.1.0 (branch: , revision: )
  build user:       
  build date:       2020-12-28T04:00:40Z
  go version:       go1.15.6
  platform:         linux/amd64
```

```
$ logcli --help |& head -n 3
2020-12-28 15:55:00.227004 I | proto: duplicate proto type registered: ingester.Series
usage: logcli [<flags>] <command> [<args> ...]

```

**Expected behavior**

No warning about duplicate protos.

**Environment:**
 - OS: Archlinux

**Comment by lngphp** [comment]
Aside of "duplicate proto", I have multiple API calls for a single command invocation where `start` parameter has the same value, but `end` is different:

`# logcli-linux-amd64 query '{app_name=~"Web.*-PROD", job="IIS"}' --limit 10000 --timezone=UTC --from '2020-12-30T03:00:00Z' --to '2020-12-30T04:00:00Z' | wc -l
2020-12-31 12:14:18.763262 I | proto: duplicate proto type registered: ingester.Series
http://localhost:3100/loki/api/v1/query_range?direction=BACKWARD&end=1609300800000000000&limit=1000&query=%7Bapp_name%3D~%22Web.%2A-PROD%22%2C+job%3D%22IIS%22%7D&start=1609297200000000000
Common labels: {filename="C:\\LogFiles\\access.log", job="IIS"}
http://localhost:3100/loki/api/v1/query_range?direction=BACKWARD&end=1609300629115175401&limit=1000&query=%7Bapp_name%3D~%22Web.%2A-PROD%22%2C+job%3D%22IIS%22%7D&start=1609297200000000000
Common labels: {filename="C:\\LogFiles\\access.log", job="IIS"}
http://localhost:3100/loki/api/v1/query_range?direction=BACKWARD&end=1609300479141045101&limit=1000&query=%7Bapp_name%3D~%22Web.%2A-PROD%22%2C+job%3D%22IIS%22%7D&start=1609297200000000000
Common labels: {filename="C:\\LogFiles\\access.log", job="IIS"}
http://localhost:3100/loki/api/v1/query_range?direction=BACKWARD&end=1609300328937877901&limit=1000&query=%7Bapp_name%3D~%22Web.%2A-PROD%22%2C+job%3D%22IIS%22%7D&start=1609297200000000000
Common labels: {filename="C:\\LogFiles\\access.log", job="IIS"}
http://localhost:3100/loki/api/v1/query_range?direction=BACKWARD&end=1609300152697776701&limit=1000&query=%7Bapp_name%3D~%22Web.%2A-PROD%22%2C+job%3D%22IIS%22%7D&start=1609297200000000000
Common labels: {filename="C:\\LogFiles\\access.log", job="IIS"}
http://localhost:3100/loki/api/v1/query_range?direction=BACKWARD&end=1609299912766136901&limit=1000&query=%7Bapp_name%3D~%22Web.%2A-PROD%22%2C+job%3D%22IIS%22%7D&start=1609297200000000000
Common labels: {filename="C:\\LogFiles\\access.log", job="IIS"}
http://localhost:3100/loki/api/v1/query_range?direction=BACKWARD&end=1609299729078217901&limit=1000&query=%7Bapp_name%3D~%22Web.%2A-PROD%22%2C+job%3D%22IIS%22%7D&start=1609297200000000000
Common labels: {filename="C:\\LogFiles\\access.log", job="IIS"}
http://localhost:3100/loki/api/v1/query_range?direction=BACKWARD&end=1609299552627697401&limit=1000&query=%7Bapp_name%3D~%22Web.%2A-PROD%22%2C+job%3D%22IIS%22%7D&start=1609297200000000000
Common labels: {filename="C:\\LogFiles\\access.log", job="IIS"}
http://localhost:3100/loki/api/v1/query_range?direction=BACKWARD&end=1609299399073452001&limit=1000&query=%7Bapp_name%3D~%22Web.%2A-PROD%22%2C+job%3D%22IIS%22%7D&start=1609297200000000000
Common labels: {filename="C:\\LogFiles\\access.log", job="IIS"}
http://localhost:3100/loki/api/v1/query_range?direction=BACKWARD&end=1609299201961448701&limit=1000&query=%7Bapp_name%3D~%22Web.%2A-PROD%22%2C+job%3D%22IIS%22%7D&start=1609297200000000000
Common labels: {filename="C:\\LogFiles\\access.log", job="IIS"}
http://localhost:3100/loki/api/v1/query_range?direction=BACKWARD&end=1609299012747483201&limit=227&query=%7Bapp_name%3D~%22Web.%2A-PROD%22%2C+job%3D%22IIS%22%7D&start=1609297200000000000
Common labels: {filename="C:\\LogFiles\\access.log", job="IIS"}
`
As I see, limit is reduced to 1000 and it looks like it does pagination, but should not both parameters change to get a data slices?

**Comment by daurnimator** [comment]
I think this was fixed with d1fbc2bc0775f7156c2cccbc3c44dbed8abd862c via #3217

**Comment by yakob-aleksandrovich** [comment]
This problem still occurs with the latest version of logcli
